src/lib.rs

```diff
@@ -0,0 +1,657 @@
+                "SNSR_RIGHT_WHEEL_EXTENDED" => status.sensor_right_wheel_extended = field.value,
+                "LSIDEBIT" => status.left_sidebit = field.value,
+                "LFRONTBIT" => status.left_frontbit = field.value,
+                "LLDSBIT" => status.left_ldsbit = field.value,
+                "RSIDEBIT" => status.right_sidebit = field.value,
+                "RFRONTBIT" => status.right_frontbit = field.value,
+                "RLDSBIT" => status.right_ldsbit = field.value,
+                _ => log::error!("Unrecognized field: {:?}", field),
+            }
+        }
+        self.digital_sensor_status = status;
+        log::debug!("Got digital_sensors");
+        Ok(status)
+    }
+
+    fn get_charger(&mut self) -> Result<ChargerStatus, GetDataError> {
+        log::debug!("get_charger");
+
+        writeln!(self.serial_port, "getcharger\n")?;
+
+        let _s = match self.read_line() {
+            Ok(v) => println!("{}", v),
+            Err(_) => println!("Error reading back"),
+        };
+
+        self.serial_port.flush()?;
+
+        log::debug!("Serial port flushed");
+
+        let mut status = ChargerStatus {
+            ..Default::default()
+        };
+
+        log::debug!("Reading values...");
+        loop {
+            let header = self.read_line()?;
+            log::debug!("{}", header);
+
+            if header.contains("Label,Value") {
+                break;
+            }
+        }
+
+        for _n in 1..15 {
+            // 15 fields
+            let s = self.read_line()?;
+            log::debug!("{}", s);
+            let _ = match IntField::from_str(s.as_str()){
+                Ok(field) => {
+                    log::debug!("{:?}", field);
+                    match field.name.as_str() {
+                        "FuelPercent" => status.fuel_percent = field.value,
+                        "BatteryOverTemp" => status.battery_over_tmp = field.value,
+                        "ChargingActive" => status.charging_active = field.value,
+                        "ChargingEnabled" => status.charging_enabled = field.value,
+                        "ConfidentOnFuel" => status.confident_on_fuel = field.value,
+                        "OnReservedFuel" => status.on_reserved_fuel = field.value,
+                        "EmptyFuel" => status.empty_fuel = field.value,
+                        "BatteryFailure" => status.battery_failure = field.value,
+                        "ExtPwrPresent" => status.ext_pwr_present = field.value,
+                        "ThermistorPresent" => status.thermistor_present = field.value,
+                        "BattTempCAvg" => status.batt_temp_c_avg = field.value,
+                        "Charger_mAH" => status.charger_mah = field.value,
+                        "Discharge_mAH" => status.discharge_mah = field.value,
+                        _ => log::error!("Unrecognized field: {:?}", field),
+                    }
+                }
+                Err(ParseIntError) => {
```

Comment:
Meh, this is kinda ugly, but it works. We know the type of each field so we can parse accordingly. Split `<T>Field` into label and value+unit maybe? So that the parser can first lookup the name, determine the type of field and then move on.

Reply:
Unit can also just be a string, so that a Field consists of `Field<T> { label: str, value: Value<T>}` and a separate `Value<T> {value: T, unit: str}`